presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -397,51 +407,72 @@ public IPrestoSparkQueryExecution create(
-            TableWriteInfo tableWriteInfo = getTableWriteInfo(session, fragmentedPlan);
-
-            JavaSparkContext javaSparkContext = new JavaSparkContext(sparkContext);
-            CollectionAccumulator<SerializedTaskInfo> taskInfoCollector = new CollectionAccumulator<>();
-            taskInfoCollector.register(sparkContext, Option.empty(), false);
-            CollectionAccumulator<PrestoSparkShuffleStats> shuffleStatsCollector = new CollectionAccumulator<>();
-            shuffleStatsCollector.register(sparkContext, Option.empty(), false);
-            TempStorage tempStorage = tempStorageManager.getTempStorage(storageBasedBroadcastJoinStorage);
-            queryStateTimer.endAnalysis();
-
-            return new PrestoSparkQueryExecution(
-                    javaSparkContext,
-                    session,
-                    queryMonitor,
-                    taskInfoCollector,
-                    shuffleStatsCollector,
-                    prestoSparkTaskExecutorFactory,
-                    executorFactoryProvider,
-                    queryStateTimer,
-                    warningCollector,
-                    sql,
-                    planAndMore,
-                    fragmentedPlan,
-                    sparkQueueName,
-                    taskInfoCodec,
-                    sparkTaskDescriptorJsonCodec,
-                    queryStatusInfoJsonCodec,
-                    queryDataJsonCodec,
-                    rddFactory,
-                    tableWriteInfo,
-                    transactionManager,
-                    createPagesSerde(blockEncodingManager),
-                    executionExceptionFactory,
-                    queryTimeout,
-                    queryCompletionDeadline,
-                    metadataStorage,
-                    queryStatusInfoOutputLocation,
-                    queryDataOutputLocation,
-                    tempStorage,
-                    nodeMemoryConfig,
-                    waitTimeMetrics);
+            Optional<QueryType> queryType = StatementUtils.getQueryType(preparedQuery.getStatement().getClass());
+
+            if (queryType.isPresent() && (queryType.get() == QueryType.DATA_DEFINITION)) {
+                DataDefinitionTask<?> task = (DataDefinitionTask<?>) ddlTasks.get(preparedQuery.getStatement().getClass());
+                queryStateTimer.endAnalysis();
+                QueryStateMachine queryStateMachine = QueryStateMachine.begin(
+                        sql,
+                        session,
+                        URI.create("http://fake.invalid/query/" + session.getQueryId()),
+                        new ResourceGroupId("test"),
+                        queryType,
+                        StatementUtils.isTransactionControlStatement(preparedQuery.getStatement()),
+                        transactionManager,
+                        accessControl,
+                        directExecutor(),
```

Comment:
Why direct executor? Usually when an executor service is requested there is an expectation it's going to provide a dedicated thread.